### DISCUSSION TITLE
Discord Auth - include more data

### DISCUSSION THREAD

**Discussion by reineling** [pocketbase/pocketbase#6143] [Q&A]
Hi! I'm working on a project, the user data i get into the db is not enough for it, i need to have discord id and much more..
Any solution to this? I'm a little new to pb.

Thanks in advance!

**Comment by chnfyi** [comment]
Hi, from my understanding you authenticate users with OAuth2 and you doesn't get enough user data for your needs.

If so, you have to fetch user data after the first (_or each regarding your use case_) successful authentication.

**Not tested example:**

```javascript
async function fetchUserData(token) {
  try {
    const response = await fetch('https://discord.com/api/users/@me', {
      headers: {
        Authorization: `Bearer ${token}`
      }
    });

    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const userData = await response.json();
    return userData
  } catch (error) {
    console.error('Error:', error);
  }
}

const authData = await pb.collection('users').authWithOAuth2({ provider: 'discord' });
const { accessToken } = authData.meta
const { id } = await fetchUserData(accessToken)
```

If you need more data like current application details you can try the `https://discord.com/api/oauth2/@me` endpoint. Check the documentation [here](https://discord.com/developers/docs/topics/oauth2#get-current-authorization-information).

**Reply by ganigeorgiev** [reply]
Manually sending request to the Discord `@me` API shouldn't be necessary.

You can specify custom scopes with the `authWithOAuth2` call and have access to the rawUser OAuth2 data in the returned response under the `meta` property, which later can be used to send an `update` call for your users if you want to fill it with some of the non-standard provider's data. For example:

```js
// authenticate
const authData = await pb.collection('users').authWithOAuth2({
    provider: 'discord',
    // specify custom scopes if needed (this vary for each OAuth2 provider)
    // scopes: ['identify', 'email'],
});
console.log(authData.meta.rawUser)

// update the user with some of the extracted rawUser fields
// (pb.authStore.record.id has the same value as authData.record.id after the above call)
await pb.collection('users').update(authData.record.id, {
   ...
});
```

**Reply by reineling** [reply]
Fetching to receive more data from the api is unnecessary as it already fetched pretty much everything.
The question was reflected more if it is possible to get more data from the function authWithOAuth2.

Instead of doing a update after the auth, maybe add something like this?
```
const authData = await pb.collection('users').authWithOAuth2({
    provider: 'discord',
    scopes: ['identify', 'email'],
    extraFields: ['discord_id', 'etc']
});
```

**Reply by ganigeorgiev** [reply]
> Instead of doing a update after the auth, maybe do something like this?

No, it is not supported (_auto mapping is possible only for the standardized fields like id, name, email, avatarURL, etc._).
